var/spack/repos/builtin/packages/protobuf/package.py

```diff
@@ -31,6 +31,7 @@ class Protobuf(AutotoolsPackage):
     homepage = "https://developers.google.com/protocol-buffers"
     url      = "https://github.com/google/protobuf/releases/download/v2.5.0/protobuf-2.5.0.tar.bz2"
 
+    version('3.1.0', '14a532a7538551d5def317bfca41dace', url='https://github.com/google/protobuf/archive/v3.1.0.tar.gz')
```

Comment:
it looks like we can create `url` for version and use `https://github.com/google/protobuf/archive/vX.Y.Z.tar.gz` for anything newer than `2.5.0` and `https://github.com/google/protobuf/releases/download/vX.Y.Z/protobuf-X.Y.Z.tar.bz2` for `2.5.0`. Do you want to give it a try?